flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/preprocess/Standardizer.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.preprocess
+
+import breeze.linalg._
+import org.apache.flink.api.scala._
+import org.apache.flink.ml.common.{Parameter, ParameterMap, Transformer}
+import org.apache.flink.ml.math.Breeze._
+import org.apache.flink.ml.math.{Vector}
+import org.apache.flink.ml.preprocess.Standardizer.{ScaleMean, ScaleStd}
+
+
+/** Scales observations, so that all features have mean equal to zero
+  * and standard deviation equal to one
+  *
+  * This transformer takes a a Vector of values and maps it into the
+  * scaled Vector that each feature has mean zero and standard deviation equal to one.
+  */
+class Standardizer extends Transformer[Vector, Vector] with Serializable {
+
+  def setScaleMean(wm: Boolean): Standardizer = {
+    parameters.add(ScaleMean, wm)
+    this
+  }
+
+  def setScaleStd(std: Boolean): Standardizer = {
+    parameters.add(ScaleStd, std)
+    this
+  }
+
+  override def transform(input: DataSet[Vector], parameters: ParameterMap):
+  DataSet[Vector] = {
+    val resultingParameters = this.parameters ++ parameters
+    val sMean = resultingParameters(ScaleMean)
+    val sStd = resultingParameters(ScaleStd)
+
+    input.map {
```

Comment:
A map operation won't be enough to calculate the mean and the standard deviation of a ```DataSet``` of ```Vector```. Each vector represents one datapoint consisting of several features. In order to calculate the mean, for example, for every feature, you have to sum up the vectors and divide by the number of vectors. This gives you a vector of feature means. You can do this with a reduce operation. Having the mean and std, you can then broadcast this value to a ```RichMapFunction``` and scale the vectors accordingly.